packages/stream-node/reader.js

```diff
@@ -0,0 +1,39 @@
+// @ts-check
+/// <reference types="ses"/>
+
+// This module provided for sake of fewer head scratches.
+// Node.js readable streams satisfy the signature of an async iterable iterator.
+// They however iterate Node.js Buffer values and are not hardened, so this
+// implementation compensates for both.
+
+/**
+ * @param {NodeJS.ReadableStream} input the source Node.js reader
+ * @returns {import('@endo/stream').Reader<Uint8Array, void>}
+ */
+export const makeNodeReader = input => {
+  const iterator = input[Symbol.asyncIterator]();
```

Comment:
Is input already an iterable-iterator? If so, why don't we just use it as `iterator` rather than calling its @@asyncIterator() method?

I note that our @@asyncIterator method does not call the underlying. I'm confused. Let's discuss.

Reply:
Node streams are async-iterable, but they're not async-iterators.

Reply:
As Mathieu answered, Node.js streams are iterable but not iterators, so this is a necessary step.